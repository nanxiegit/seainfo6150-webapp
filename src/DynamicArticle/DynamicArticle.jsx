import React from "react";
import HTMLText from "../HTMLText/HTMLText";

const DynamicArticle = (props) => {
  return (
    <article>
      {/* add extra html with javascript attributes here */}
      <header>
        <h1>You can take a pie on a plane</h1>
        <address>
          by Cathryn Lindsey (
          <a href="mailto:cathryn.lindsey@nytimes.com">cathryn.lindsey@nytimes.com</a>)
          <br />
        </address>
        <time datetime="2018-11-21 14:57:55">November 21st 2018, 2:57 pm </time>
      </header>
      <HTMLText text={props.article.text} />
    </article>
  );
};

export default DynamicArticle;
