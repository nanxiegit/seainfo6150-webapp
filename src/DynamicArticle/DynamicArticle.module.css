.article {
    background: #fff;
    border: 1px dashed #666;
    border-radius: 10px;
    box-shadow: 0px 0px 10px 5px rgba(204, 0, 0, 0.5);
    font-family: Arial, Helvetica, sans-serif;
    list-style-type: none;
    padding: 10px; 
  }

.back {
    display: block;
    color: #c00;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 20px;
    font-weight: bold;
    margin-bottom: 20px;
}

.container{
  column-width: 200px;
  column-rule: 1px solid rgba(255, 255, 255, 0.3);
  column-gap: 2em;
  margin: 0 auto;
  max-width: 960px;
  border-radius: 2px;
  padding: 15px;

}

