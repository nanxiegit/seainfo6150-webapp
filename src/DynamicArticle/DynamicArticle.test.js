
import React from "react";
import DynamicArticle from "./DynamicArticle";

describe("DynamicArticle tests", () => {
  it("renders correctly", () => {
    const article = {
        title: "You can take a pie on a plane",
        image: {
          "_url": "https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
        },
    };
    const { container } = render(<DynamicArticle  article={article} />);
    expect(container).toMatchSnapshot();
  });
});