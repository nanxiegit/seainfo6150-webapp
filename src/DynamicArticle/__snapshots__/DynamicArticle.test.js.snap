// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicArticle tests renders correctly 1`] = `
<div>
  <div>
    <a
      class="back"
      href="/articlelist"
    >
      Back
    </a>
    <article
      class="article"
    >
      <header>
        <img
          alt="You can take a pie on a plane"
          class="image"
          src="https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
        />
        <h1>
          You can take a pie on a plane
        </h1>
        <address>
          by 
           (
          <a
            href="mailto:undefined"
          />
          )
          <br />
        </address>
        <time />
      </header>
      <article
        class="container"
      >
        <div />
      </article>
    </article>
  </div>
</div>
`;
